samcli/commands/_utils/custom_options/hook_package_id_option.py

```diff
@@ -28,73 +33,91 @@ def __init__(self, *args, **kwargs):
-                return super().handle_parse_result(ctx, opts, args)
-
-            # call prepare hook
-            built_template_path = DEFAULT_BUILT_TEMPLATE_PATH
-            if not self._force_prepare and os.path.exists(built_template_path):
-                LOG.info("Skip Running Prepare hook. The current application is already prepared.")
-            else:
-                LOG.info("Running Prepare Hook to prepare the current application")
-
-                iac_project_path = os.getcwd()
-                output_dir_path = os.path.join(iac_project_path, ".aws-sam-iacs", "iacs_metadata")
-                if not os.path.exists(output_dir_path):
-                    os.makedirs(output_dir_path, exist_ok=True)
-                debug = opts.get("debug", False)
-                aws_profile = opts.get("profile")
-                aws_region = opts.get("region")
-                metadata_file = iac_hook_wrapper.prepare(
-                    output_dir_path, iac_project_path, debug, aws_profile, aws_region
+                    f"Parameters {opt_name}, and {','.join(self._invalid_coexist_options)} cannot be used together"
                 )
 
-                LOG.info("Prepare Hook is done, and metadata file generated at %s", metadata_file)
-                opts["template_file"] = metadata_file
-        return super().handle_parse_result(ctx, opts, args)
+
+def _validate_build_command_parameters(command_name, opts):
+    # validate build-image is provided in case of build using container
+    # add this validation here to avoid running hook prepare and there is issue
+    if command_name == "build" and opts.get("use_container") and not opts.get("build_image"):
+        raise click.UsageError("Missing required parameter --build-image.")
+
+
+def _check_experimental_flag(hook_package_id, command_name, opts):
+    # check beta-feature
+    beta_features = opts.get("beta_features")
+
+    # check if beta feature flag is required for a specific hook package
+    # The IaCs support experimental flag map will contain only the beta IaCs. In case we support the external
+    # hooks, we need to first know that the hook package is an external, and to handle the beta feature of it
+    # using different approach
+    experimental_entry = ExperimentalFlag.IaCsSupport.get(hook_package_id)
+    if beta_features is None and experimental_entry is not None:
+        iac_support_message = _get_iac_support_experimental_prompt_message(hook_package_id, command_name)
+        if not prompt_experimental(experimental_entry, iac_support_message):
+            LOG.debug("Experimental flag is disabled and prepare hook is not run")
+            return False
+    elif not beta_features:
+        LOG.debug("--beta-feature flag is disabled and prepare hook is not run")
+        return False
+    elif beta_features:
+        LOG.debug("--beta-feature flag is enabled, update the experimental flag.")
```

Comment:
`--beta-features` instead of `--beta-feature`

Reply:
Reword the sentence: --beta-features flag is enabled, enabling <xyz> experimental flag.